---
title: Interfaces
layout: page
navigation: true
sub_title: Use PopPUNK
order: 2
---

{% include page_banner.html %}
<section class="blog-area themeix-ptb">
   <div class="container">
       <div class="row">
           <!-- Start Blog Content -->
           <div class="col-md-9">
               <div class="blog-wrapper">
                     <div class="blog-content-desc">

                        <p>A full description of the method can be found in the <a href="https://doi.org/10.1101/gr.241455.118">paper</a>.</p>

                        <p><code class="highlighter-rouge">PopPUNK</code> uses the fast k-mer distance estimation enabled by <a href="https://mash.readthedocs.io/en/latest/&gt;">mash</a>
                        to calculate core and accessory distances between all pairs of isolates of bacteria in a collection. By clustering
                        these distances into 'within-strain' and 'between-strain' distances a network
                        of within-strain comparisons can be constructed. The use of a network has
                        a number of convenient properties, the first being that the connected
                        components represent a cluster of strains.</p>

                        <p>As well as identifying strains, the pairwise distance distribution also helps
                        with assembly quality control (particularly in the case of contaminated
                        contigs) and may be informative of the level of recombination in the
                        population. The network representation also allows definition of representative isolates by
                        sampling one example from each clique, and calculation of various statistics
                        which can show how good the clustering is.</p>

                        <p>The advantages of this approach are broadly that:</p>

                        <ul>
                           <li> It is fast, and scalable to 10^4 genomes in a single run.
                           <li>Assigning new query sequences to a cluster using an existing database is scalable even beyond this.
                           <li>Databases can be updated online (as sequences arrive).
                           <li>Online updating is equivalent to building databases from scratch.
                           <li>Databases can be kept small and managable by only keeping representative isolates.
                           <li>There is no bin cluster. Outlier isolates will be in their own cluster.
                           <li>Pre-processing, such as generation of an alignment, is not required.
                           <li>The definition of clusters is biologically relevant to how bacteria evolve.
                           <li>There is a lot of quantitative and graphical output to assist with clustering.
                           <li>A direct import into  <a href="https://microreact.org/">microreact</a> is
                           available, as well as <a href="http://www.cytoscape.org/&gt;">cytoscape</a>,
                           <a href="http://dx.doi.org/10.1101/gr.232397.117&gt;">grapetree</a> and
                           <a href="http://jameshadfield.github.io/phandango/">phandango</a>.</li>
                           <li>Everything is available within a single python executable.</li>
                        </ul>

                        <h2 id="citation">Citation:</h2>
                        <p>If you find PopPUNK useful, please cite as:</p>

                        <p>Lees JA, Harris SR, Tonkin-Hill G, Gladstone RA, Lo SW, Weiser JN, Corander J, Bentley SD, Croucher NJ. Fast and flexible
                        bacterial genomic epidemiology with PopPUNK. <em>Genome Research</em> <strong>29</strong>:1-13 (2019).
                        doi:<a href="https://dx.doi.org/10.1101/gr.241455.118">10.1101/gr.241455.118</a></p>
                     </div>
               </div>
            </div>
         </div>
      </div>
   </section>

